Services/communication_service.c

```diff
@@ -35,31 +36,309 @@
+      }
+      S_FS.Firmware_Version = csp_htonflt(S_FS.Firmware_Version);
+
+      memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+      memcpy(&packet->data[OUT_DATA_BYTE], &S_FS, sizeof(S_FS));
+      set_packet_length(packet, sizeof(int8_t) + sizeof(S_FS) + 1);
+      if (queue_response(packet) != SATR_OK) {
+        return SATR_ERROR;
+      }
+      break;
+
+      case SET_FREQ:
+          cnv8_F(&packet->data[IN_DATA_BYTE], &S_config.freq);
+          //S_config.freq = (float) packet->data[IN_DATA_BYTE];
+          S_config.freq = csp_ntohflt(S_config.freq);
+          status = HAL_S_setFreq(S_config.freq);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) +  1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+
+      case SET_PA_POWER:
+          S_config.PA_Power = (uint8_t)packet->data[IN_DATA_BYTE];
+          S_config.PA_Power = csp_ntoh32((uint32_t)S_config.PA_Power);
+          status = HAL_S_setPAPower(S_config.PA_Power);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) + 1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case SET_CONTROL:
+          S_config.PA.status = (uint8_t)packet->data[IN_DATA_BYTE];
+          S_config.PA.mode = (uint8_t)packet->data[IN_DATA_BYTE + 1];
+          S_config.PA.status = csp_ntoh32((uint32_t)S_config.PA.status);
+          S_config.PA.mode = csp_ntoh32((uint32_t)S_config.PA.mode);
+          status = HAL_S_setControl(S_config.PA);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) +  1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case SET_ENCODER:
+          S_config.enc.scrambler = (uint8_t)packet->data[IN_DATA_BYTE];
+          S_config.enc.filter = (uint8_t)packet->data[IN_DATA_BYTE + 1];
+          S_config.enc.modulation = (uint8_t)packet->data[IN_DATA_BYTE + 2];
+          S_config.enc.rate = (uint8_t)packet->data[IN_DATA_BYTE + 3];
+          S_config.enc.scrambler = csp_ntoh32((uint32_t)S_config.enc.scrambler);
+          S_config.enc.filter = csp_ntoh32((uint32_t)S_config.enc.filter);
+          S_config.enc.modulation = csp_ntoh32((uint32_t)S_config.enc.modulation);
+          S_config.enc.rate = csp_ntoh32((uint32_t)S_config.enc.rate);
+          status = HAL_S_setEncoder(S_config.enc);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) +  1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case SET_CONFIG:
+          cnv8_F(&packet->data[IN_DATA_BYTE], &S_config.freq);
```

Comment:
Now, (even when I get back to this version in the PR), `S_config.freq` doesn't get the value I send from GS. And the problem happens here even when I update the HAL function. As far as I remember, there was no such inconsistency between `set_freq` and `set_config`.
I've double-checked the GS code. It seems fine. I even took the float entry to the end in case the 4 bytes for the float is messing sth up but it didn't solve anything. Another weird thing was that although the frequency float data was the last one in the packet, it seemed that still this part of the code was remained untouched, meaning that other parameters were receiving the wrong value this time (even though I changed the `[IN_DATA_BYTE + 6]` stuff, accordingly).  So, I'm not sure if this was even working correctly.

Reply:
We can get rid of this functionality though. It's just sth that sets everything with one command, saving some time every time we want to do a test. However, it is alarming that sth could be wrong that we can't see and might come up later when other stuff are also added.